tests/test_sources.py

```diff
@@ -97,3 +111,50 @@ def test___init___parent(self, mocker: MockerFixture, tmp_path: Path) -> None:
         assert obj.toml_table_header == ('some', 'table')
         assert obj.toml_data == {'field': 'some'}
         assert obj.toml_file_path == tmp_path / 'pyproject.toml'
+
+
+@pytest.mark.skipif(azure_key_vault is None, reason='azure-keyvault-secrets and azure-identity are not installed')
+class TestAzureKeyVaultSettingsSource:
+    """Test AzureKeyVaultSettingsSource."""
+
+    def test___init__(self) -> None:
+        """Test __init__."""
+
+        class AzureKeyVaultSettings(BaseSettings):
+            """AzureKeyVault settings."""
+
+        AzureKeyVaultSettingsSource(
+            AzureKeyVaultSettings, 'https://my-resource.vault.azure.net/', DefaultAzureCredential()
+        )
+
+    def test_get_field_value(self, mocker: MockerFixture) -> None:
+        """Test _get_field_value."""
+
+        class AzureKeyVaultSettings(BaseSettings):
+            """AzureKeyVault settings."""
+
+        key_vault_secret = KeyVaultSecret(SecretProperties(), 'SecretValue')
+        mocker.patch(f'{MODULE}.SecretClient.get_secret', return_value=key_vault_secret)
+        obj = AzureKeyVaultSettingsSource(
+            AzureKeyVaultSettings, 'https://my-resource.vault.azure.net/', DefaultAzureCredential()
+        )
+
+        obj.get_field_value(field=FieldInfo(), field_name='sqlserverpassword')
```

Comment:
Don't we need any assertion here?

Reply:
The method is just a mocked HTTP call to an API. I think I can't test anything more. The reliable test is the Jupyter notebook I'm executing against my Azure infrastructure.

Reply:
You can test the returned value from the mocked method.

I think it would be good to have a complete test like other tests. I mean having a settings model that uses the `AzureKeyVaultSettingsSource`.



```py
def test_azure_key_vault_settings_source():
    class AzureKeyVaultSettings(BaseSettings):
        my_password: str
        sql_server__password: str
        @classmethod
        def settings_customise_sources(
            cls,
            settings_cls: type[BaseSettings],
            init_settings: PydanticBaseSettingsSource,
            env_settings: PydanticBaseSettingsSource,
            dotenv_settings: PydanticBaseSettingsSource,
            file_secret_settings: PydanticBaseSettingsSource,
        ) -> tuple[PydanticBaseSettingsSource, ...]:
            return (
                AzureKeyVaultSettingsSource(
                    settings_cls, os.environ['KEY_VAULT__URL'], DefaultAzureCredential()
                ),
            )

    with mocker.patch(f'{MODULE}.SecretClient.get_secret', return_value=<return value>):
        settings = AzureKeyVaultSettings()
        assert settings.my_password == ...
        assert settings.sql_server__password == ...```

Reply:
Done.